<html><h3>45d7c856ed0bdf76c3b351249ad3e27c75bdeeef,cnvlib/segfilters.py,,cn,#Any#,112
</h3><link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
@require_column(&quotcn&quot)
def cn(segarr):
    Merge segments by integer copy number.
    <a id="change">return segarr</a>


@require_column(&quotsem&quot)
def sem(segarr):</code></pre><h3>After Change</h3><pre><code class='java'>
def cn(segarr):
    Merge segments by integer copy number.
    groups = enumerate_changes(segarr[&quotcn&quot], segarr[&quotchromosome&quot])
    data = <a id="change">segarr</a>.data.assign(group=groups)
    if &quotcn1&quot in segarr:
        &#47&#47 Keep allele-specific CNAs separate
        data[&quotg1&quot] = enumerate_changes(segarr[&quotcn1&quot])
        data[&quotg2&quot] = enumerate_changes(segarr[&quotcn2&quot])
        <a id="change">groupkey = [&quotgroup&quot, &quotg1&quot, &quotg2&quot]</a>
    else:
        groupkey = &quotgroup&quot
    data = data.groupby(groupkey, sort=False, as_index=False, group_keys=False
                       ).apply(squash_region)
    <a id="change">return segarr.as_dataframe(data)</a>


@require_column(&quotsem&quot)
def sem(segarr):</code></pre><img src="128850254.png" alt="Italian Trulli"   style="width:500px;height:500px;"><div id='inPattern'>In pattern: SUPERPATTERN</div><BR><div id='frequency'>Frequency: 3</div><BR><div id='size'>Non-data size: 5</div><BR><h3>Instances</h3><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/45d7c856ed0bdf76c3b351249ad3e27c75bdeeef#diff-9932ec4387c7317b25d4e4d06c021298e320b79c9ebce339e42fa7d69290adb2L114' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 45d7c856ed0bdf76c3b351249ad3e27c75bdeeef</div><div id='time'> Time: 2016-09-01</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/segfilters.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: cn</div><BR><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/45d7c856ed0bdf76c3b351249ad3e27c75bdeeef#diff-9932ec4387c7317b25d4e4d06c021298e320b79c9ebce339e42fa7d69290adb2L114' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 45d7c856ed0bdf76c3b351249ad3e27c75bdeeef</div><div id='time'> Time: 2016-09-01</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/segfilters.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: cn</div><BR><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/03cd68309bcb94d8bd760621998cfc851a683fac#diff-9932ec4387c7317b25d4e4d06c021298e320b79c9ebce339e42fa7d69290adb2L44' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 03cd68309bcb94d8bd760621998cfc851a683fac</div><div id='time'> Time: 2016-09-01</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/segfilters.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: ampdel</div><BR><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/85cb4379eae497129d8944a8fd2fab3cd34b2dc2#diff-b5fd1371a8e7979cf9de89d7a81e1001805e759b0d1fda7fcca52389739ca290L542' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 85cb4379eae497129d8944a8fd2fab3cd34b2dc2</div><div id='time'> Time: 2016-11-21</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/genome/gary.py</div><div id='class'> Class Name: GenomicArray</div><div id='method'> Method Name: subtract</div><BR>