<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
            dframe[&quotend&quot] = dframe[&quotend_cds&quot].astype(&quotint&quot)
        else:
            dframe[&quotstart&quot] = dframe[&quotstart&quot].astype(&quotint&quot) - 1
            <a id="change">dframe[&quotend&quot]</a> = dframe[&quotend&quot].astype(&quotint&quot)

    for key in (&quotstart_cds&quot, &quotend_cds&quot, &quotexon_starts&quot, &quotexon_ends&quot):
        <a id="change">del dframe[key]</a>
    &#47&#47 NB: same gene name can appear on alt. contigs
    dframe = (dframe.groupby(by=[&quotchromosome&quot, &quotstrand&quot, &quotgene&quot],
                             as_index=False, group_keys=False, sort=False)
              .apply(_merge_overlapping))</code></pre><h3>After Change</h3><pre><code class='java'>

    cols_shared = [&quotgene&quot, &quotaccession&quot, &quotchromosome&quot, &quotstrand&quot]
    if exons:
        cols_rest = <a id="change">[&quot_start_tx&quot, &quot_end_tx&quot,  &#47&#47 Transcription
                     &quot_start_cds&quot, &quot_end_cds&quot,  &#47&#47 Coding region
                     &quotexon_count&quot, &quotexon_starts&quot, &quotexon_ends&quot]</a>
    elif cds:
        &#47&#47 Use CDS instead of transcription region
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,
                     &quotstart&quot, &quotend&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    else:
        cols_rest = [&quotstart&quot, &quotend&quot,
                     &quot_start_cds&quot, &quot_end_cds&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    <a id="change">colnames = cols_shared + cols_rest</a>
    usecols = [c <a id="change">for</a> c in colnames if not c.startswith(&quot_&quot)]
    &#47&#47 Parse the file contents
    try:
        dframe = pd.read_table(infile,  header=None,</code></pre>