<html><h3>9a9778a6d9f4b2751a7ff5e0a2d4dcd7e6eee0a2,rnaseq/src/run_STAR.py,,,#,20
</h3><link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
else:
    with open(args.fastq1) as f:
        f.readline()
        seq = <a id="change">f</a>.readline().strip()
        read_length = len(seq)
<a id="change">overhang = read_length-1</a>

if read_length&lt;=201:
    starcmd=&quotSTAR&quot
else:</code></pre><h3>After Change</h3><pre><code class='java'>
parser.add_argument(&quot--outSAMattributes&quot, default=[&quotNH&quot, &quotHI&quot, &quotAS&quot, &quotnM&quot, &quotNM&quot, &quotch&quot], nargs=&quot+&quot)
parser.add_argument(&quot--chimSegmentMin&quot, default=&quot15&quot, help=&quotMinimum chimeric segment length; switches on detection of chimeric (fusion) alignments&quot)
parser.add_argument(&quot--chimJunctionOverhangMin&quot, default=&quot15&quot, help=&quotMinimum overhang for a chimeric junction&quot)
parser.add_argument(&quot--chimOutType&quot, default=<a id="change">[&quotWithinBAM&quot, &quotSoftClip&quot]</a>, nargs=&quot+&quot, help=&quot&quot)
parser.add_argument(&quot--chimMainSegmentMultNmax&quot, default=&quot1&quot, help=&quot&quot)
parser.add_argument(&quot--genomeLoad&quot, default=&quotNoSharedMemory&quot)
parser.add_argument(&quot--STARlong&quot, action=&quotstore_true&quot, help=&quotUse STARlong instead of STAR&quot)</code></pre><img src="27432963.png" alt="Italian Trulli"   style="width:500px;height:500px;"><div id='inPattern'>In pattern: SUPERPATTERN</div><BR><div id='frequency'>Frequency: 3</div><BR><div id='size'>Non-data size: 4</div><BR><h3>Instances</h3><BR><div id='link'><a href='https://github.com/broadinstitute/gtex-pipeline/commit/9a9778a6d9f4b2751a7ff5e0a2d4dcd7e6eee0a2#diff-5f7afe23e6147a3f8ca14ffdfc1fa791a7533d7c6cdb8f772af787d714ca25ceL20' target='_blank'>Link</a></div><div id='project'> Project Name: broadinstitute/gtex-pipeline</div><div id='commit'> Commit Name: 9a9778a6d9f4b2751a7ff5e0a2d4dcd7e6eee0a2</div><div id='time'> Time: 2017-04-11</div><div id='author'> Author: francois@broadinstitute.org</div><div id='file'> File Name: rnaseq/src/run_STAR.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: </div><BR><BR><div id='link'><a href='https://github.com/pantsbuild/pants/commit/2649b5de8d60e813bc842c9bfe8b4e7b222c49ad#diff-da5814ec23e6d37add917305985c7589d48c0a04e9d56faa29da1ea022263fbdL343' target='_blank'>Link</a></div><div id='project'> Project Name: pantsbuild/pants</div><div id='commit'> Commit Name: 2649b5de8d60e813bc842c9bfe8b4e7b222c49ad</div><div id='time'> Time: 2021-03-12</div><div id='author'> Author: john.sirois@gmail.com</div><div id='file'> File Name: src/python/pants/backend/python/util_rules/pex_test.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: create_pex_and_get_all_data</div><BR><BR><div id='link'><a href='https://github.com/QUANTAXIS/QUANTAXIS/commit/619499d6bacb220833b73731e550120af7bdee94#diff-fcfb0360c920d3bb3583f505d017f18fdb59ce0f24d9e568240a53e6155f5d46L9' target='_blank'>Link</a></div><div id='project'> Project Name: QUANTAXIS/QUANTAXIS</div><div id='commit'> Commit Name: 619499d6bacb220833b73731e550120af7bdee94</div><div id='time'> Time: 2017-03-23</div><div id='author'> Author: yutiansut@qq.com</div><div id='file'> File Name: analysis/python/example.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: </div><BR>