<html><h3>4d934160aa7f4f9f1d5cdea7cd0e66cfc3d43d05,third_party/nucleus/io/fasta.py,InMemoryRefReader,__init__,#InMemoryRefReader#Any#,130
</h3><link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
      if start &lt; 0:
        raise ValueError(&quotstart={} must be &gt;= for chromosome={}&quot.format(
            start, contig_name))
      <a id="change">if contig_name in self._chroms:
        raise ValueError(&quotDuplicate chromosome={} detect&quot.format(contig_name))
     </a> if not bases:
        raise ValueError(
            &quotBases must contain at least one base, but got "{}"&quot.format(bases))

      end = start + len(bases)
      <a id="change">self._chroms[contig_name] = _InMemoryChromosome(start, end, bases)</a>
      contigs.append(
          reference_pb2.ContigInfo(
              name=contig_name, n_bases=end, pos_in_fasta=i))
</code></pre><h3>After Change</h3><pre><code class='java'>
    
    super(InMemoryRefReader, self).__init__()

    <a id="change">ref_seqs = []</a>
    contigs = []
    for i, (contig_name, start, bases) in enumerate(chromosomes):
      if start &lt; 0:
        raise ValueError(&quotstart={} must be &gt;= for chromosome={}&quot.format(</code></pre><img src="28471250.png" alt="Italian Trulli"   style="width:500px;height:500px;"><div id='inPattern'>In pattern: SUPERPATTERN</div><BR><div id='frequency'>Frequency: 3</div><BR><div id='size'>Non-data size: 6</div><BR><h3>Instances</h3><BR><div id='link'><a href='https://github.com/google/deepvariant/commit/4d934160aa7f4f9f1d5cdea7cd0e66cfc3d43d05#diff-0af9e0e95f32e9761ad1209c60eb15e5556245472df0c298301687dd044940bbL126' target='_blank'>Link</a></div><div id='project'> Project Name: google/deepvariant</div><div id='commit'> Commit Name: 4d934160aa7f4f9f1d5cdea7cd0e66cfc3d43d05</div><div id='time'> Time: 2018-04-29</div><div id='author'> Author: mdepristo@google.com</div><div id='file'> File Name: third_party/nucleus/io/fasta.py</div><div id='class'> Class Name: InMemoryRefReader</div><div id='method'> Method Name: __init__</div><BR><BR><div id='link'><a href='https://github.com/wkentaro/labelme/commit/7ad8d5c5cf89269d191117c4dd41ce205bc66c8c#diff-20a434242a3644257f0c023ee8b47787fb12f85bd0dfeab0e3c059660b076bc3L149' target='_blank'>Link</a></div><div id='project'> Project Name: wkentaro/labelme</div><div id='commit'> Commit Name: 7ad8d5c5cf89269d191117c4dd41ce205bc66c8c</div><div id='time'> Time: 2019-05-15</div><div id='author'> Author: www.kentaro.wada@gmail.com</div><div id='file'> File Name: labelme/widgets/label_dialog.py</div><div id='class'> Class Name: LabelDialog</div><div id='method'> Method Name: resetFlags</div><BR><BR><div id='link'><a href='https://github.com/OpenNMT/OpenNMT-py/commit/e2799c64b7b8248de872ca7fb741080079839710#diff-004ff622c411caa26172d7c3ca873f7011649e0573e40798b4ce022426fe6444L648' target='_blank'>Link</a></div><div id='project'> Project Name: OpenNMT/OpenNMT-py</div><div id='commit'> Commit Name: e2799c64b7b8248de872ca7fb741080079839710</div><div id='time'> Time: 2019-02-11</div><div id='author'> Author: dylan.flaute@gmail.com</div><div id='file'> File Name: onmt/translate/translator.py</div><div id='class'> Class Name: Translator</div><div id='method'> Method Name: _translate_batch_deprecated</div><BR>