<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
      if start &lt; 0:
        raise ValueError(&quotstart={} must be &gt;= for chromosome={}&quot.format(
            start, contig_name))
      <a id="change">if contig_name in self._chroms:
        raise ValueError(&quotDuplicate chromosome={} detect&quot.format(contig_name))
     </a> if not bases:
        raise ValueError(
            &quotBases must contain at least one base, but got "{}"&quot.format(bases))
</code></pre><h3>After Change</h3><pre><code class='java'>
    
    super(InMemoryRefReader, self).__init__()

    <a id="change">ref_seqs = []</a>
    contigs = []
    for i, (contig_name, start, bases) in enumerate(chromosomes):
      if start &lt; 0:
        raise ValueError(&quotstart={} must be &gt;= for chromosome={}&quot.format(
            start, contig_name))
      if not bases:
        raise ValueError(
            &quotBases must contain at least one base, but got "{}"&quot.format(bases))

      end = start + len(bases)
      ref_seqs.append(reference_pb2.ReferenceSequence(
          region=ranges.make_range(contig_name, start, end), bases=bases))
      contigs.append(
          reference_pb2.ContigInfo(
              name=contig_name, n_bases=end, pos_in_fasta=i))

    self._reader = <a id="change">fasta_reader.InMemoryGenomeReference.create(
        contigs, ref_seqs)</a>
    self.header = RefFastaHeader(contigs=self._reader.contigs)

  def iterate(self):
    raise NotImplementedError(&quotCan not iterate through a FASTA file&quot)</code></pre>