<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
P_reference.add_argument(&quotreferences&quot, nargs=&quot*&quot,
        help=Normal-sample target or antitarget .cnn files, or the
                directory that contains them.)
<a id="change">P_reference.add_argument(&quot-f&quot, &quot--fasta&quot,
        help="Reference genome, FASTA format (e.g. UCSC hg19.fa)")</a>
P_reference.add_argument(&quot-o&quot, &quot--output&quot, help="Output file name.")
P_reference.add_argument(&quot-y&quot, &quot--male-reference&quot, action=&quotstore_true&quot,
        help=Create a male reference: shift female samples&quot chrX
                log-coverage by -1, so the reference chrX average is -1.
                Otherwise, shift male samples&quot chrX by +1, so the reference chrX
                average is 0.)
<a id="change">P_reference.add_argument(&quot-x&quot, &quot--sample-sex&quot, &quot-g&quot, &quot--gender&quot,
        dest=&quotsample_sex&quot,
        choices=(&quotm&quot, &quoty&quot, &quotmale&quot, &quotMale&quot, &quotf&quot, &quotx&quot, &quotfemale&quot, &quotFemale&quot),
        help=Specify the chromosomal sex of all given samples as male or
                female. (Default: guess each sample from coverage of X and Y
                chromosomes).)</a>

P_reference_flat = P_reference.add_argument_group(
    "To construct a generic, \"flat\" copy number reference with neutral "
    "expected coverage")
P_reference_flat.add_argument(&quot-t&quot, &quot--targets&quot,
        help="Target intervals (.bed or .list)")
P_reference_flat.add_argument(&quot-a&quot, &quot--antitargets&quot,
        help="Antitarget intervals (.bed or .list)")

<a id="change">P_reference_bias = P_reference.add_argument_group(
    "To disable specific automatic bias corrections")</a>
<a id="change">P_reference_bias.add_argument(&quot--no-gc&quot, dest=&quotdo_gc&quot, action=&quotstore_false&quot,
        help="Skip GC correction.")</a>
<a id="change">P_reference_bias.add_argument(&quot--no-edge&quot, dest=&quotdo_edge&quot, action=&quotstore_false&quot,
        help="Skip edge-effect correction.")</a>
<a id="change">P_reference_bias.add_argument(&quot--no-rmask&quot, dest=&quotdo_rmask&quot, action=&quotstore_false&quot,
        help="Skip RepeatMasker correction.")</a>
P_reference.set_defaults(func=_cmd_reference)


&#47&#47 fix -------------------------------------------------------------------------</code></pre><h3>After Change</h3><pre><code class='java'>


P_coverage = AP_subparsers.add_parser(&quotcoverage&quot, help=_cmd_coverage.__doc__)
<a id="change">P_coverage.add_argument(&quotbam_file&quot, help="Mapped sequence reads (.bam)")</a>
<a id="change">P_coverage.add_argument(&quotinterval&quot, help="Intervals (.bed or .list)")</a>
<a id="change">P_coverage.add_argument(&quot-c&quot, &quot--count&quot, action=&quotstore_true&quot,
        help=Get read depths by counting read midpoints within each bin.
                (An alternative algorithm).)</a>
<a id="change">P_coverage.add_argument(&quot-q&quot, &quot--min-mapq&quot, type=int, default=0,
        help=Minimum mapping quality score (phred scale 0-60) to count a read
                for coverage depth.  [Default: %(default)s])</a>
P_coverage.add_argument(&quot-o&quot, &quot--output&quot, help=Output file name.)
<a id="change">P_coverage.add_argument(&quot-p&quot, &quot--processes&quot,
        nargs=&quot?&quot, type=int, const=0, default=1,
        help=Number of subprocesses to calculate coverage in parallel.
                Without an argument, use the maximum number of available CPUs.
                [Default: use 1 process])</a>
P_coverage.set_defaults(func=_cmd_coverage)


&#47&#47 reference -------------------------------------------------------------------</code></pre>