<html><h3>1fe319ef49168c2c7c8d10a7c426c1414521d164,cnvlib/tabio/genepred.py,,read_refflat,#Any#Any#Any#,80
</h3><link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
        &#47&#47 dframe.apply,applymap,...
        raise NotImplementedError
    else:
        <a id="change">dframe = dframe.sort_values([&quotchromosome&quot, &quotstart&quot, &quotend&quot])</a>
        <a id="change">dframe[&quotstart&quot] -= 1</a>

    &#47&#47 NB: same gene name can appear on alt. contigs
    <a id="change">dframe = (dframe.groupby(by=[&quotchromosome&quot, &quotstrand&quot, &quotgene&quot],
                             as_index=False, group_keys=False, sort=False)
              .apply(_merge_overlapping))</a>
    return dframe


def _merge_overlapping(dframe):</code></pre><h3>After Change</h3><pre><code class='java'>
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,  &#47&#47 Transcription
                     &quot_start_cds&quot, &quot_end_cds&quot,  &#47&#47 Coding region
                     &quot_exon_count&quot, &quotexon_starts&quot, &quotexon_ends&quot]
        <a id="change">converters = {&quotexon_starts&quot: _split_commas, &quotexon_ends&quot: _split_commas}</a>
    elif cds:
        &#47&#47 Use CDS instead of transcription region
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,
                     &quotstart&quot, &quotend&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    else:
        cols_rest = [&quotstart&quot, &quotend&quot,
                     &quot_start_cds&quot, &quot_end_cds&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    colnames = cols_shared + cols_rest
    usecols = [c for c in colnames if not c.startswith(&quot_&quot)]
    &#47&#47 Parse the file contents
    dframe = pd.read_table(infile,  header=None, na_filter=False,
                           names=colnames, usecols=usecols,
                           dtype={c: str for c in cols_shared},
                           converters=converters)

    &#47&#47 Calculate values for output columns
    if exons:
        <a id="change">dframe = pd.DataFrame.from_records(_split_exons(dframe),
                                           columns=cols_shared + [&quotstart&quot, &quotend&quot])</a>
        <a id="change">dframe[&quotstart&quot] = dframe[&quotstart&quot].astype(&quotint&quot)</a>
        dframe[&quotend&quot] = dframe[&quotend&quot].astype(&quotint&quot)

    return (dframe.assign(start=dframe.start - 1)
            .sort_values([&quotchromosome&quot, &quotstart&quot, &quotend&quot])</code></pre><img src="306064268.png" alt="Italian Trulli"   style="width:500px;height:500px;"><div id='inPattern'>In pattern: SUPERPATTERN</div><BR><div id='frequency'>Frequency: 3</div><BR><div id='size'>Non-data size: 8</div><BR><h3>Instances</h3><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/1fe319ef49168c2c7c8d10a7c426c1414521d164#diff-12c9ac77cdf2ffa7b67be9ce77ec9b63738695673a4e30d5b0c46d6be3198b40L115' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 1fe319ef49168c2c7c8d10a7c426c1414521d164</div><div id='time'> Time: 2016-10-28</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/tabio/genepred.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: read_refflat</div><BR><BR><div id='link'><a href='https://github.com/NTMC-Community/MatchZoo/commit/53a93d8af5d041f6c5dfdef2fb763f8419e3023d#diff-041d63e4d0745af739dfa9a00dd27b00615fce44b3d14d04c3bfb2636beec936L35' target='_blank'>Link</a></div><div id='project'> Project Name: NTMC-Community/MatchZoo</div><div id='commit'> Commit Name: 53a93d8af5d041f6c5dfdef2fb763f8419e3023d</div><div id='time'> Time: 2019-01-19</div><div id='author'> Author: i@uduse.com</div><div id='file'> File Name: matchzoo/datasets/snli/load_data.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: load_data</div><BR><BR><div id='link'><a href='https://github.com/bashtage/linearmodels/commit/8698f9b95638a258c880e0ae25cdf8859f26bfa3#diff-57c337191c383abec54d0799ad9eeca5d48daba9fe5361c7de29d18d192b2549L682' target='_blank'>Link</a></div><div id='project'> Project Name: bashtage/linearmodels</div><div id='commit'> Commit Name: 8698f9b95638a258c880e0ae25cdf8859f26bfa3</div><div id='time'> Time: 2017-04-05</div><div id='author'> Author: kevin.k.sheppard@gmail.com</div><div id='file'> File Name: linearmodels/panel/model.py</div><div id='class'> Class Name: BetweenOLS</div><div id='method'> Method Name: _choose_cov</div><BR>