<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
        &#47&#47 dframe.apply,applymap,...
        raise NotImplementedError
    else:
        <a id="change">dframe = dframe.sort_values([&quotchromosome&quot, &quotstart&quot, &quotend&quot])</a>
        <a id="change">dframe[&quotstart&quot] -= 1</a>

    &#47&#47 NB: same gene name can appear on alt. contigs
    <a id="change">dframe = (dframe.groupby(by=[&quotchromosome&quot, &quotstrand&quot, &quotgene&quot],
                             as_index=False, group_keys=False, sort=False)
              .apply(_merge_overlapping))</a>
    return dframe


def _merge_overlapping(dframe):</code></pre><h3>After Change</h3><pre><code class='java'>
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,  &#47&#47 Transcription
                     &quot_start_cds&quot, &quot_end_cds&quot,  &#47&#47 Coding region
                     &quot_exon_count&quot, &quotexon_starts&quot, &quotexon_ends&quot]
        <a id="change">converters = {&quotexon_starts&quot: _split_commas, &quotexon_ends&quot: _split_commas}</a>
    elif cds:
        &#47&#47 Use CDS instead of transcription region
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,
                     &quotstart&quot, &quotend&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    else:
        cols_rest = [&quotstart&quot, &quotend&quot,
                     &quot_start_cds&quot, &quot_end_cds&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    colnames = cols_shared + cols_rest
    usecols = [c for c in colnames if not c.startswith(&quot_&quot)]
    &#47&#47 Parse the file contents
    dframe = pd.read_table(infile,  header=None, na_filter=False,
                           names=colnames, usecols=usecols,
                           dtype={c: str for c in cols_shared},
                           converters=converters)

    &#47&#47 Calculate values for output columns
    if exons:
        <a id="change">dframe = pd.DataFrame.from_records(_split_exons(dframe),
                                           columns=cols_shared + [&quotstart&quot, &quotend&quot])</a>
        <a id="change">dframe[&quotstart&quot] = dframe[&quotstart&quot].astype(&quotint&quot)</a>
        dframe[&quotend&quot] = dframe[&quotend&quot].astype(&quotint&quot)

    return (dframe.assign(start=dframe.start - 1)
            .sort_values([&quotchromosome&quot, &quotstart&quot, &quotend&quot])</code></pre>