<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
    &#47&#47 Reformat this data to create INFO and genotype
    &#47&#47 TODO be more clever about this
    for out_row, abs_row in zip(out_dframe.itertuples(index=False),
                                <a id="change">abs_dframe.itertuples(index=False)</a>):
        if (out_row.ncopies == abs_row.expect or
            &#47&#47 Survive files from buggy v0.7.1 (&#47&#4753)
            not str(out_row.probes).isdigit()):</code></pre><h3>After Change</h3><pre><code class='java'>
    Convert copy number segments to VCF records.
    out_dframe = segments.data.loc[:, ["chromosome", "end", "log2", "probes"]]
    if "cn" in segments:
        out_dframe["ncopies"] = <a id="change">segments["cn"]</a>
        abs_expect = call.absolute_expect(segments, ploidy, is_sample_female)

    else:
        abs_dframe = call.absolute_dataframe(segments, ploidy, 1.0,</code></pre>