<html><h3>1fe319ef49168c2c7c8d10a7c426c1414521d164,cnvlib/tabio/genepred.py,,read_refflat,#Any#Any#Any#,80
</h3><link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
    dframe = (dframe.groupby(by=[&quotchromosome&quot, &quotstrand&quot, &quotgene&quot],
                             as_index=False, group_keys=False, sort=False)
              .apply(_merge_overlapping))
    <a id="change">return dframe</a>


def _merge_overlapping(dframe):
    Merge overlapping regions within a group.</code></pre><h3>After Change</h3><pre><code class='java'>
        raise ValueError("Arguments &quotcds&quot and &quotexons&quot are mutually exclusive")

    cols_shared = [&quotgene&quot, &quotaccession&quot, &quotchromosome&quot, &quotstrand&quot]
    <a id="change">converters = None</a>
    if exons:
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,  &#47&#47 Transcription
                     &quot_start_cds&quot, &quot_end_cds&quot,  &#47&#47 Coding region
                     &quot_exon_count&quot, &quotexon_starts&quot, &quotexon_ends&quot]
        converters = {&quotexon_starts&quot: _split_commas, &quotexon_ends&quot: _split_commas}
    elif cds:
        &#47&#47 Use CDS instead of transcription region
        cols_rest = [&quot_start_tx&quot, &quot_end_tx&quot,
                     &quotstart&quot, &quotend&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    else:
        cols_rest = [&quotstart&quot, &quotend&quot,
                     &quot_start_cds&quot, &quot_end_cds&quot,
                     &quot_exon_count&quot, &quot_exon_starts&quot, &quot_exon_ends&quot]
    colnames = cols_shared + cols_rest
    usecols = [c for c in colnames if not c.startswith(&quot_&quot)]
    &#47&#47 Parse the file contents
    dframe = pd.read_table(infile,  header=None, na_filter=False,
                           names=colnames, usecols=usecols,
                           dtype={c: str for c in cols_shared},
                           converters=converters)

    &#47&#47 Calculate values for output columns
    if exons:
        <a id="change">dframe = pd.DataFrame.from_records(_split_exons(dframe),
                                           columns=cols_shared + [&quotstart&quot, &quotend&quot])</a>
        dframe[&quotstart&quot] = dframe[&quotstart&quot].astype(&quotint&quot)
        dframe[&quotend&quot] = dframe[&quotend&quot].astype(&quotint&quot)

    <a id="change">return (dframe.assign(start=dframe.start - 1)
            .sort_values([&quotchromosome&quot, &quotstart&quot, &quotend&quot])
            .reset_index(drop=True))</a>


def _split_commas(field):
    return field.rstrip(&quot,&quot).split(&quot,&quot)</code></pre><img src="252525369.png" alt="Italian Trulli"   style="width:500px;height:500px;"><div id='inPattern'>In pattern: SUPERPATTERN</div><BR><div id='frequency'>Frequency: 3</div><BR><div id='size'>Non-data size: 5</div><BR><h3>Instances</h3><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/1fe319ef49168c2c7c8d10a7c426c1414521d164#diff-12c9ac77cdf2ffa7b67be9ce77ec9b63738695673a4e30d5b0c46d6be3198b40L115' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 1fe319ef49168c2c7c8d10a7c426c1414521d164</div><div id='time'> Time: 2016-10-28</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/tabio/genepred.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: read_refflat</div><BR><BR><div id='link'><a href='https://github.com/metagenome-atlas/atlas/commit/3ab0e54a3ce88c7018a34192461dd47e5867d357#diff-6572a918b48fd81b22f482976cf3565e6e14b936a5f62ddb8d8593781ca543acL61' target='_blank'>Link</a></div><div id='project'> Project Name: metagenome-atlas/atlas</div><div id='commit'> Commit Name: 3ab0e54a3ce88c7018a34192461dd47e5867d357</div><div id='time'> Time: 2017-01-28</div><div id='author'> Author: joe.brown@pnnl.gov</div><div id='file'> File Name: atlas/tables.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: col_split</div><BR><BR><div id='link'><a href='https://github.com/etal/cnvkit/commit/11d76fd4eb2ef5ecb3c062ff4ff4f39d64ae5741#diff-012276feb06260bb4553906b1a9eab37580d1372d4f535a317ec19cb9ad7e4d6L20' target='_blank'>Link</a></div><div id='project'> Project Name: etal/cnvkit</div><div id='commit'> Commit Name: 11d76fd4eb2ef5ecb3c062ff4ff4f39d64ae5741</div><div id='time'> Time: 2016-11-24</div><div id='author'> Author: eric.talevich@gmail.com</div><div id='file'> File Name: cnvlib/genome/_merge.py</div><div id='class'> Class Name: </div><div id='method'> Method Name: _merge</div><BR>