<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>

P_export_bed = P_export_subparsers.add_parser(&quotbed&quot,
        help=_cmd_export_bed.__doc__)
<a id="change">P_export_bed.add_argument(&quotsegments&quot, nargs=&quot+&quot,
        help=Segmented copy ratio data files (*.cns), the output of the
                &quotsegment&quot or &quotcall&quot sub-commands.)</a>
<a id="change">P_export_bed.add_argument("-i", "--sample-id", metavar="LABEL",
        help=Identifier to write in the 4th column of the BED file.
                [Default: use the sample ID, taken from the file name])</a>
<a id="change">P_export_bed.add_argument("--ploidy", type=int, default=2,
        help="Ploidy of the sample cells. [Default: %(default)d]")</a>
<a id="change">P_export_bed.add_argument(&quot-x&quot, &quot--sample-sex&quot, &quot-g&quot, &quot--gender&quot,
        dest=&quotsample_sex&quot,
        choices=(&quotm&quot, &quoty&quot, &quotmale&quot, &quotMale&quot, &quotf&quot, &quotx&quot, &quotfemale&quot, &quotFemale&quot),
        help=Specify the sample&quots chromosomal sex as male or female.
                (Otherwise guessed from X and Y coverage).)</a>
<a id="change">P_export_bed.add_argument("--show",
        choices=(&quotploidy&quot, &quotvariant&quot, &quotall&quot), default="ploidy",
        help=Which segmented regions to show:
                &quotall&quot = all segment regions;
                &quotvariant&quot = CNA regions with non-neutral copy number;
                &quotploidy&quot = CNA regions with non-default ploidy.
                [Default: %(default)s])</a>
<a id="change">P_export_bed.add_argument("-y", "--male-reference", action="store_true",
        help=Was a male reference used?  If so, expect half ploidy on
                chrX and chrY; otherwise, only chrY has half ploidy.  In CNVkit,
                if a male reference was used, the "neutral" copy number (ploidy)
                of chrX is 1; chrY is haploid for either reference sex.)</a>
P_export_bed.add_argument(&quot-o&quot, &quot--output&quot, help="Output file name.")
P_export_bed.set_defaults(func=_cmd_export_bed)

</code></pre><h3>After Change</h3><pre><code class='java'>
        nargs=&quot+&quot, metavar="FILES",
        help=Tabular files with Ensembl gene ID and number of reads mapped to
                each gene, from RSEM or another transcript quantifier.)
<a id="change">P_import_rna.add_argument(&quot-f&quot, &quot--format&quot,
        choices=(&quotrsem&quot, &quotcounts&quot), default=&quotcounts&quot, metavar=&quotNAME&quot,
        help=Input format name: &quotrsem&quot for RSEM gene-level read counts
                (*_rsem.genes.results), or &quotcounts&quot for generic 2-column gene
                IDs and their read counts (e.g. TCGA level 2 RNA expression).
                )</a>
P_import_rna.add_argument(&quot-g&quot, &quot--gene-resource&quot, metavar="FILE",
        help="Location of gene info table from Ensembl BioMart.")
<a id="change">P_import_rna.add_argument(&quot-c&quot, &quot--correlations&quot, metavar="FILE",
        help=Correlation of each gene&quots copy number with
        expression. Output of cnv_expression_correlate.py.)</a>
<a id="change">P_import_rna.add_argument(&quot-d&quot, &quot--output-dir&quot,
        default=&quot.&quot, metavar="PATH",
        help=Directory to write a CNVkit .cnr file for each input
                sample. [Default: %(default)s])</a>
P_import_rna.add_argument(&quot-o&quot, &quot--output&quot, metavar="FILE",
        help="Output file name (summary table).")
P_import_rna.set_defaults(func=_cmd_import_rna)
</code></pre>