<link rel="stylesheet" href="../../../../default.css">
<script src="../../../../highlight.pack.js"></script> 
<script>hljs.initHighlightingOnLoad();</script>
<html><h3></h3><h3>Before Change</h3><pre><code class='java'>
            dv_call.variant, sam_reader=sample.in_memory_sam_reader)
        for sample in self.samples[samples_id]
    ]
    <a id="change">if self.options.pic_options.alt_aligned_pileup != &quotnone&quot:
      &#47&#47 Align the reads against each alternate allele, saving the sequences of
      &#47&#47 those alleles along with the alignments for pileup images.
      alt_info_for_samples = [
          self.align_to_all_haplotypes(dv_call.variant, reads)
          for reads in reads_for_samples
      ]
      &#47&#47 Each sample has different reads and thus different alt-alignments.
      haplotype_alignments_for_samples = [
          sample[&quotalt_alignments&quot] for sample in alt_info_for_samples
      ]
      &#47&#47 All samples share the same alt sequences, so select the first one.
      haplotype_sequences = alt_info_for_samples[0][&quotalt_sequences&quot]
      pileup_images = self.pics[samples_id].create_pileup_images(
          dv_call=dv_call,
          reads_for_samples=reads_for_samples,
          haplotype_alignments_for_samples=haplotype_alignments_for_samples,
          haplotype_sequences=haplotype_sequences)
    else:
      pileup_images = self.pics[samples_id].create_pileup_images(
          dv_call=dv_call, reads_for_samples=reads_for_samples)

   </a> if pileup_images is None:
      &#47&#47 We cannot build a PileupImage for dv_call, issue a warning.
      logging.warning(&quotCould not create PileupImage for candidate at %s:%s&quot,
                      dv_call.variant.reference_name, dv_call.variant.start)</code></pre><h3>After Change</h3><pre><code class='java'>
        for sample in self.samples[samples_id]
    ]
    &#47&#47 Decide whether each candidate needs ALT-alignment.
    <a id="change">alt_align_this_variant = False</a>
    <a id="change">if self.options.pic_options.alt_aligned_pileup != &quotnone&quot:
      if self.options.pic_options.types_to_alt_align == &quotindels&quot:
        alt_align_this_variant = variant_utils.is_indel(dv_call.variant)
      else:  &#47&#47 types_to_alt_align can only be &quotall&quot or &quotindels&quot.
        alt_align_this_variant = True

   </a> haplotype_alignments_for_samples = None
    haplotype_sequences = None
    <a id="change">if alt_align_this_variant:
      &#47&#47 Align the reads against each alternate allele, saving the sequences of
      &#47&#47 those alleles along with the alignments for pileup images.
      alt_info_for_samples = [
          self.align_to_all_haplotypes(dv_call.variant, reads)
          for reads in reads_for_samples
      ]
      &#47&#47 Each sample has different reads and thus different alt-alignments.
      haplotype_alignments_for_samples = [
          sample[&quotalt_alignments&quot] for sample in alt_info_for_samples
      ]

      &#47&#47 All samples share the same alt sequences, so select the first one.
      haplotype_sequences = alt_info_for_samples[0][&quotalt_sequences&quot]

   </a> pileup_images = self.pics[samples_id].create_pileup_images(
        dv_call=dv_call,
        reads_for_samples=reads_for_samples,
        haplotype_alignments_for_samples=haplotype_alignments_for_samples,</code></pre>